go/obscuronode/host/client.go

```diff
@@ -24,20 +24,21 @@ type EnclaveRPCClient struct {
 	protoClient generated.EnclaveProtoClient
 	connection  *grpc.ClientConn
 	timeout     time.Duration
+	nodeID      common.Address
 }
 
-func NewEnclaveRPCClient(address string, timeout time.Duration) *EnclaveRPCClient {
+func NewEnclaveRPCClient(address string, timeout time.Duration, nodeID common.Address) *EnclaveRPCClient {
 	opts := []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}
 	connection, err := grpc.Dial(address, opts...)
 	if err != nil {
-		panic(fmt.Sprintf("failed to connect to enclave RPC service: %v", err))
+		panic(fmt.Errorf(">   Agg%d: Failed to connect to enclave RPC service: %w", obscurocommon.ShortAddress(nodeID), err))
```

Comment:
We should really have a Stringer implementation instead of using this ShortAddress formatter.

Reply:
A stringer? In any case, would rather pick it up in a separate PR. We use this everywhere.

Reply:
Something like this perhaps ? https://go.dev/tour/methods/17

Fair point 👍  I'll catalog it for a future PR

Reply:
The problem was that not all of these were our types. I agree , we should have stringers where possible. Let's raise a ticket

Reply:
Added a task on the backlog 👍